src/amperity/gocd/secret/vault/plugin.clj

```diff
@@ -137,10 +138,22 @@
 ;; request body will also have the configuration required to connect and lookup
 ;; for secrets from the external Secret Manager.
 (defmethod handle-request "go.cd.secrets.secrets-lookup"
-  [_ _ data]
-  ;; TODO: See https://plugin-api.gocd.org/19.7.0/secrets/#lookup-secrets for desired response
-  (let [configuration (:configuration data)
-        secret-keys (:keys data)]
-    {:response-code    500
-     :response-headers {}
-     :response-body    "NYI"}))
+  [client _ data]
+  (try
+    (let [secret-keys (:keys data)
+          secrets (mapv (fn [key]
+                          {:key   key
+                           :value (vault/read-secret client key {:not-found nil})})
+                        secret-keys)
+          filtered_secret_keys (mapv :key (filter #(nil? (:value %)) secrets))]
```

Comment:
Style: use kebab-case for names, not snake_case. It's also not clear from this name that this is a list of keys which were not found. Combining some suggestions:
```suggestion
          missing-keys (mapv :key (remove :value secrets))]
```